src/notification-banner/private/_NotificationBannerView.jsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Button from 'terra-button';
+import IconAlert from 'terra-icon/lib/icon/IconAlert';
+import IconError from 'terra-icon/lib/icon/IconError';
+import IconWarning from 'terra-icon/lib/icon/IconWarning';
+import IconGapChecking from 'terra-icon/lib/icon/IconGapChecking';
+import IconDiamondSymbol from 'terra-icon/lib/icon/IconDiamondSymbol';
+import IconInformation from 'terra-icon/lib/icon/IconInformation';
+import IconSuccess from 'terra-icon/lib/icon/IconSuccess';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import ThemeContext from 'terra-theme-context';
+
+import { ApplicationIntlContext } from '../../application-intl';
+
+import useElementSize, { breakpointFilter } from './useElementSize';
+
+import styles from './NotificationBannerView.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const NotificationTypes = {
+  ALERT: 'alert',
+  ERROR: 'error',
+  WARNING: 'warning',
+  UNSATISFIED: 'unsatisfied',
+  UNVERIFIED: 'unverified',
+  ADVISORY: 'advisory',
+  INFO: 'info',
+  SUCCESS: 'success',
+  CUSTOM: 'custom',
+};
+
+const propTypes = {
+  /**
+   * An action element to be added to the action section of the alert to give the user an easy way
+   * to accomplish a task to resolve the notification.
+   */
+  action: PropTypes.element,
+  /**
+   * Child Nodes providing the message content for the alert. Can contain text and HTML.
+   */
+  children: PropTypes.node,
+  /**
+   * The icon to be used for an alert of type custom. This will not be used for any other alert types.
+   */
+  customIcon: PropTypes.element,
+  /**
+   * Sets an author-defined class, to control the status bar color to be used for an alert of type custom.
+   *
+   * ![IMPORTANT](https://badgen.net/badge//IMPORTANT/CSS?icon=github)
+   * Adding `var(--my-app...` CSS variables is required for proper re-themeability when creating custom color styles _(see included examples)_.
+   */
+  customColorClass: PropTypes.string,
```

Comment:
do we still support custom colors and icons through this api?

Reply:
We do. We have talked offline about potentially removing support for custom banners in v2.